### ISSUE TITLE
Array remove shouldn't print an error when index out of bounds (for consistency with actual documentation).

### ISSUE THREAD

**Issue by just-like-that** [godotengine/godot-proposals#5192]
### Describe the project you are working on

Making any game/app where an array is involved.

### Describe the problem or limitation you are having in your project

The documentation of Array.remove(position: int) states:

Removes an element from the array by index. 
**If the index does not exist in the array, nothing happens.** 
To remove an element by searching for its value, use erase() instead.

Actually something happens in the case of for instance:
`ids.remove(-1)`

There is an error generated with a reference to the underlying C++ source code - in debug mode at least.
https://github.com/godotengine/godot/blob/419e713a29f20bd3351a54d1e6c4c5af7ef4b253/core/cowdata.h#L164

Exceeding the array index at the other end results in the same error.

```

	_FORCE_INLINE_ void remove(int p_index) {
// => cause of error message - line below:
		ERR_FAIL_INDEX(p_index, size()); 
		T *p = ptrw();
		int len = size();
		for (int i = p_index; i < len - 1; i++) {
			p[i] = p[i + 1];
		};

		resize(len - 1);
	};

```

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Array.remove(position) should not print errors if the index is out of bounds.

Suppress the error message to be consistent with pop_front and pop_back, which do not print errors.

[Array front and back shouldn't print an error when empty (for consistency with pop_front and pop_back).
#5117](https://github.com/godotengine/godot-proposals/issues/5117)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Works like nowadays just without an error being generated on each call which floods the error console.

`Array.remove(position) `

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Will be used often and can NOT be worked around by switching off as Debug-warnings.

### Is there a reason why this should be core and not an add-on in the asset library?

It's a core feature.

**Comment by zinnschlag** [comment]
IMO these proposals are moving into the wrong direction. Consistency is valuable. But changes should be made instead to always report errors (and change the documentation accordingly).

A quote from the Zen of Python:

> Errors should never pass silently.
> Unless explicitly silenced.

This does not only apply to Python, but it a good guideline in general.

**Comment by YuriSizov** [comment]
I agree with @zinnschlag. If your code floods the output log with errors, as you say, there is an issue with your code. Silencing error reporting on the engine level is not helpful and is going to create problems for other users.

At most we could add a warning ignore annotation for such cases, but this is not marked as a warning, but as an error. Because it is an error. In user code.

**Comment by just-like-that** [comment]
> At most we could add a warning ignore annotation for such cases, but this is not marked as a warning, but as an error. Because it is an error. In user code.

In my scenario it's not an error. It's done in accordance with the current documentation.

The documentation explicitly states:
**If the index does not exist in the array, nothing happens.**

That's the reason why I omitted an extra check which is done in the engines code anyway  as pointed out.

So whether it's a change in the documentation OR a change in the engine code which I personally prefer
otherwise it leads to redundant check inside and outside of the function call.

Just do nothing if a none existing index is to be removed as with erase(). 
Erase is also silent in case that no item was removed (hence not found).

**Comment by YuriSizov** [comment]
> In my scenario it's NOT an error. It's done in accordance with the current documentation.

Then the documentation should be changed.
